### PR TITLE
testcafe tests

### PR DESCRIPTION
**Pull request by miherlosev**

### CODE REVIEW

File: test/e2e/index-test.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import IndexPage from './page-model';
```
Comment (cdroulers) [inline test/e2e/index-test.js]: Rename to "main.password.spec.js" to follow convention for other tests.

main.{fixture}.spec.js for all tests, to separate concerns.

File: test/e2e/page-model.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { Selector } from 'testcafe';
```
Comment (cdroulers) [inline test/e2e/page-model.js]: Rename to `IndexPage.js` and move to a `pages` folder as well.

**Review by cdroulers — CHANGES_REQUESTED** [review]
If have minor changes I put as comments.

Also, use 2 spaces instead of 4 and an lf at the end of the file (to respect `.editorconfig` settings)

Then I'll be happy to merge!

I can also merge in another branch and to that myself!

**Comment by cdroulers** [comment]
Wow! That was quick and easy! I'm impressed with how easy that was!

I had used the closed source testcafé before and thought it was ok, but this is pretty amazing!

I just installed nightmare and the plugin to test it and the test fails with 

```
× enter password

   1) A call to an async function is not awaited. Use the "await" keyword before actions, assertions or chains of them to ensure
      that they run in the right sequence.

      Browser: Electron 1.4.15 / Windows 10 0.0.0

         14 |        .typeText(indexPage.dialogs.enterPassword.password, 'wrong')
         15 |        .click(indexPage.dialogs.enterPassword.submit)
         16 |        .expect(indexPage.messageBar.text.textContent).contains("Couldn't find a level with password wrong.")
         17 |        .click(indexPage.menu.enterPassword)
         18 |        .click(indexPage.dialogs.enterPassword.cancel)
       > 19 |        .expect(indexPage.dialogs.enterPassword.visible).notOk();
         20 |});
         21 |

         at <anonymous> (C:\dev\cdroulers\block-fly\test\e2e\index-test.js:19:57)
         at step (C:\dev\cdroulers\block-fly\test\e2e\index-test.js:1:612)
         at <anonymous> (C:\dev\cdroulers\block-fly\test\e2e\index-test.js:1:782)
```

I tried to add a debugger statement in the `test('enter password'` but it didn't stop. Any pointers?

File: test/e2e/index-test.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import IndexPage from './page-model';
+
+fixture `Left menu`
+    .page('http://localhost:5555/');
+
+const indexPage = new IndexPage();
+
+test('enter password', async t => {
```
Comment (cdroulers) [inline test/e2e/index-test.js]: Use double-quotes! I forgot about that. I'll setup eslint later.

**Comment by miherlosev** [comment]
Cann't reproduce the issue with nightmare.
See [video](https://www.screencast.com/t/JYdJZowcaTM) with my actions.
It looks like nightmare problem as in Google Chrome all works good.

**Comment by miherlosev** [comment]
You can learn about all features related to debugging in https://github.com/VasilyStrelyaev/testcafe/blob/4d3ca2b7962dae68bf57778417bbdab8a7ca4cc7/docs/articles/documentation/recipes/debugging-tests.md

This article is in progress.
At present, you can use everything this article suggests except for the `t.debug` action - It will be avalible in the next public release.

**Review by cdroulers — APPROVED** [review]

**Comment by miherlosev** [comment]
Hello @cdroulers.
As I see, you have a good impression about TestCafe.
Could you please tweet about us if it's possible?

**Comment by cdroulers** [comment]
I can tweet to my 2 followers, but I'll probably blog about it soon!
Thanks for the help

**Pull request merged**